### PR TITLE
feat: Implement people advanced filter drawer - EXO-60555 - Meeds-io/MIPs#39

### PR DESCRIPTION
**Pull request by sofyenne**

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<template>
+  <exo-drawer
+      ref="PeopleAdvancedFilterDrawer"
+      id="PeopleAdvancedFilterDrawer"
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
      ref="peopleAdvancedFilterDrawer"
      id="peopleAdvancedFilterDrawer"
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      right
+      @closed="close">
+    <template slot="title">
+      <span class="PopupTitle">{{ $t('pepole.advanced.filter.title') }}</span>
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
      <span class="popupTitle">{{ $t('pepole.advanced.filter.title') }}</span>
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            </template>
+          </v-btn>
+        </div>
+      </div>
+    </template>
+  </exo-drawer>
+</template>
+<script>
+export default {
+  name: 'PeopleAdvancedFilterDrawer',
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
  name: 'peopleAdvancedFilterDrawer',
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    };
+  },
+  created() {
+    this.$root.$on('open-people-advanced-filter-drawer', this.openDrawer);
+    this.getSettings();
+  },
+  watch: {
+    keyword() {
+      this.settings = this.settings.map((element) => {
+        const keywrd = this.keyword.split(' ');
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
        const splittedKeyword = this.keyword.split(' ');
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    openDrawer() {
+      this.$refs.PeopleAdvancedFilterDrawer.open();
+    },
+    close() {
+      this.$refs.PeopleAdvancedFilterDrawer.close();
+    },
+    getSettings() {
+      return this.$profileSettingsService.getSettings()
+        .then(settings => {
+          const notAlowedPrp = ['ims.skype','ims.jitsi','ims.msn','ims.facebook','ims','phones.other','phones.work','phones.home','ims.other'];
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
          const notAlowedProps = ['ims.skype','ims.jitsi','ims.msn','ims.facebook','ims','phones.other','phones.work','phones.home','ims.other'];
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+          this.settings =  settings.filter((e) => e.active && !notAlowedPrp.includes(e.propertyName) ).map(obj => ({ ...obj, valueToSearch: '' })) || [];
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
          this.settings =  settings.filter((e) => e.active && !notAlowedPrp.includes(e.propertyName)).map(obj => ({ ...obj, valueToSearch: '' })) || [];
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        });
+    },
+    getResolvedName(item){
+      const lang = eXo && eXo.env.portal.language || 'en';
+      const resolvedLabel = item.labels.find(v => v.language === lang);
+      if (resolvedLabel){
+        return resolvedLabel.label;
+      }
+      return this.$t && this.$t(`profileSettings.property.name.${item.propertyName}`)!==`profileSettings.property.name.${item.propertyName}`?this.$t(`profileSettings.property.name.${item.propertyName}`):item.propertyName;
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
      return this.$t && this.$t(`profileSettings.property.name.${item.propertyName}`) !== `profileSettings.property.name.${item.propertyName}` ?this.$t(`profileSettings.property.name.${item.propertyName}`) : item.propertyName;
```

File: webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js [hunk 8]
```diff
@@ -27,3 +35,5 @@ if (extensionRegistry) {
     });
   }
 }
+
+
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js]: remove empty lines

```suggestion
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 1 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 2 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 3 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 4 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: As discussed with PO , no need to this change !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 5 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed!

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 6 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue (hunk 7 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: fixed !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js (hunk 8 above)
Comment (sofyenne) [reply webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js]: fixed !

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    openDrawer() {
+      this.$refs.peopleAdvancedFilterDrawer.open();
+    },
+    close() {
+      this.$refs.peopleAdvancedFilterDrawer.close();
+    },
+    getSettings() {
+      return this.$profileSettingsService.getSettings()
+        .then(settings => {
+          const notAlowedProps = ['ims.skype','ims.jitsi','ims.msn','ims.facebook','ims','phones.other','phones.work','phones.home','ims.other'];
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]: to be validated with PO

File: webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties [hunk 10]
```diff
@@ -689,3 +689,13 @@ profileSettings.property.name.team=Team
...
 profileSettings.property.name.country=Country
 profileSettings.property.name.city=City
+
+#####################################################################################
+#                                    People Advanced Filter                                 #
+#####################################################################################
+pepole.advanced.filter.button.confirm=Confirm
+pepole.advanced.filter.button.cancel=Cancel
+pepole.advanced.filter.button.reset=reset
+pepole.advanced.filter.title=Filter people list
```
Comment (azayati) [inline webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties]: move theses labels to peopleList portlet properties instead

**Review by azayati — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+            type="text"
+            class="input-block-level ignore-vuetify-classes my-3">
+      </div>
+      </div>
+      </v-form>
+    </template>
+    <template slot="footer">
+      <div class="VuetifyApp flex d-flex">
+        <v-btn
+            class="resetButtonFilterPeopl" id="resetButtonFilterPeopl">
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleAdvancedFilterDrawer.vue]:
```suggestion
            class="peopleAdvancedFilterResetButton">
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js [hunk 12]
```diff
@@ -27,3 +35,4 @@ if (extensionRegistry) {
     });
   }
 }
+
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/people-list/initComponents.js]: empty line, remove it

**Review by azayati — CHANGES_REQUESTED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**